### PR TITLE
Allow setting the run name as a command-line option

### LINKED ISSUE #135: Allow setting the run name as a command-line option

### PR DESCRIPTION
**Pull request by keppy**
Fixes #135 

Changes proposed in this pull request:
- Added the "-n/--name" option to `tango run`. This option allows the user to give the experiment an arbitrary name.

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/allenai/tango/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/allenai/tango/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [x] If this PR fixes a bug, I've added a test that will fail without my fix.
- [x] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [x] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

**Comment by keppy** [comment]
I was unsure if this `name` should in fact be passed to `Executor.execute_step_graph()`. What would it be used for there if I did pass it in?

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `tango.settings` module and `tango settings` group of commands.
 - A format for storing sequences as `SqliteSparseSequence`
 - A way to massage kwargs before they determine the unique ID of a `Step`
+- Added the "-n/--name" option to `tango run`. This option allows the user to give the experiment an arbitrary name.
```
Comment (epwalsh) [inline CHANGELOG.md:25]: Can you move this up to "Unreleased"?

File: tango/__main__.py [hunk 2]
```diff
@@ -227,6 +227,12 @@ def cleanup(*args, **kwargs):
     help="Start a server that visualizes the current run",
     default=True,
 )
+@click.option(
+    "-n",
+    "--name",
+    type=str,
+    help="""Specify a name for the experiment""",
```
Comment (epwalsh) [inline tango/__main__.py:234]: How about:
```suggestion
    help="""Specify the name for this run""",
```

**Review by epwalsh — COMMENTED** [review]
Thanks @keppy! Just a couple minor comments.

> I was unsure if this name should in fact be passed to Executor.execute_step_graph(). What would it be used for there if I did pass it in?

That suggestion is out-of-date so please disregard.

**Pull request merged**